todo_app/app.py

```diff
@@ -21,7 +66,12 @@ def index():
 
     # Add a new card to the list
     @app.route('/add-card', methods=["POST"])
+    @login_required
     def new_card():
+
+        if current_user.role != "writer":
+            return "Forbidden", 403
```

Comment:
Nice - I wasn't able to bypass this. If we wanted to make it a little more reusable we might look at adding a decorator, but it's potentially just for the sake of it at this scale!